smartrunner_driver/src/rosnode/smartrunner_node.cpp

```diff
@@ -45,21 +155,21 @@ sr3dnode::sr3dnode():nh_("~")
             {
                 // Declare publisher and create timer
                 scan_publisher_         = nh_.advertise<sensor_msgs::PointCloud>("scan",100);
-                cmd_subscriber_         = nh_.subscribe("control_command",100,&sr3dnode::cmdMsgCallback,this);
-                get_scan_data_timer_    = nh_.createTimer(ros::Duration(0.1), &sr3dnode::getScanDataPointCloud, this);
+                cmd_subscriber_         = nh_.subscribe("control_command",100,&smartrunner_node::cmdMsgCallback,this);
+                get_scan_data_timer_    = nh_.createTimer(ros::Duration(0.1), &smartrunner_node::getScanDataPointCloud, this);
             }
             if(message_type_ == "PointCloud2")
             { 
                 // Declare publisher and create timer
                 scan_publisher_         = nh_.advertise<sensor_msgs::PointCloud2>("scan",100);
-                cmd_subscriber_         = nh_.subscribe("control_command",100,&sr3dnode::cmdMsgCallback,this);
-                get_scan_data_timer_    = nh_.createTimer(ros::Duration(0.1), &sr3dnode::getScanDataPointCloud2, this);
+                cmd_subscriber_         = nh_.subscribe("control_command",100,&smartrunner_node::cmdMsgCallback,this);
+                get_scan_data_timer_    = nh_.createTimer(ros::Duration(0.1), &smartrunner_node::getScanDataPointCloud2, this);
```

Comment:
I'd prefer, that the grabbing will not use a fixed 10Hz, timer but an own thread with a grab loop that will push the image whenever one becomes available.
This approach has multiple downsides:
1. additional latency
2. the auto trigger frequency basically has no effect on the actual framerate of the system